Add JIT option to enable SCC disclaiming

ymanton opened the pull request:
This option will be used by a new feature in OpenJ9 and will, at some point, be inverted to disable the functionality rather than enable.

dsouzai reviewed (commented):
> This option will be used by a new feature in OpenJ9 and will, at some point, be inverted to disable the functionality rather than enable.

Perhaps you should add `disableSharedCacheDisclaiming` with `RESET_OPTION_BIT(TR_EnableSharedCacheDisclaiming)` in this PR itself so that you don't need to open another OMR PR in the future.

ymanton commented:
> > This option will be used by a new feature in OpenJ9 and will, at some point, be inverted to disable the functionality rather than enable.
> 
> Perhaps you should add `disableSharedCacheDisclaiming` with `RESET_OPTION_BIT(TR_EnableSharedCacheDisclaiming)` in this PR itself so that you don't need to open another OMR PR in the future.

Thanks, forgot about that convention. Updated the patch and moved things around to be in alpha order.

dsouzai commented:
jenkins build xlinux
